frontend/attest/sbom.go

```diff
@@ -19,12 +19,16 @@ import (
 
 type Scanner[T any] func(ctx context.Context, resolver llb.ImageMetaResolver, target *result.Result[T], keys []string) error
 
-func CreateScanner[T any](scanner reference.Named, toState func(context.Context, T) (llb.State, error), fromState func(context.Context, llb.State) (T, error)) Scanner[T] {
+func CreateSBOMScanner[T any](scanner reference.Named, toState func(context.Context, T) (llb.State, error), fromState func(context.Context, llb.State) (T, error)) Scanner[T] {
 	if scanner == nil {
 		return nil
 	}
 
 	return func(ctx context.Context, resolver llb.ImageMetaResolver, target *result.Result[T], keys []string) error {
+		if _, ok := target.Metadata[exptypes.ExporterHasSBOM]; ok {
```

Comment:
This doesn't look very clean. Maybe we should just call it `SBOMBundle` or just add a predicate hint for the bundle.

Reply:
I like the idea of doing a predicate hint for the bundle, since then that means we can treat that as a filter to determine what attestations are allowed in a bundle (as a follow-up).